pkg/provider/azure.go

```diff
@@ -236,6 +240,11 @@ type Config struct {
 	Tags string `json:"tags,omitempty" yaml:"tags,omitempty"`
 }
 
+// HasExtendedLocation returns true if extendedlocation prop are specified.
+func (config *Config) HasExtendedLocation() bool {
+	return config.ExtendedLocationName != "" && config.ExtendedLocationType != ""
```

Comment:
`ExtendedLocationType` is redundant, if `ExtendedLocationName` is non empty, then use extended location.

Reply:
seems both config are used for LB and PIP. If so, we should check both.